lib/local_participant_view.dart

```diff
@@ -0,0 +1,130 @@
+// © 2022 Daily, Co. All Rights Reserved
+
+import 'dart:async';
+
+import 'package:daily_flutter/daily_flutter.dart';
+import 'package:daily_flutter_demo/user_settings_bottom_sheet.dart';
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class LocalParticipantView extends StatefulWidget {
+  const LocalParticipantView({super.key, required this.client, required this.prefs});
+
+  final CallClient client;
+  final SharedPreferences prefs;
+
+  @override
+  State<LocalParticipantView> createState() => _LocalParticipantViewState();
+}
+
+class _LocalParticipantViewState extends State<LocalParticipantView> {
+  StreamSubscription? _eventSubscription;
+  final _controller = VideoViewController();
+  late String _username = widget.client.participants.local.info.username ?? 'Guest';
+
+  @override
+  void initState() {
+    super.initState();
+    _controller.setTrack(widget.client.participants.local.media?.camera.track);
+    _eventSubscription = widget.client.events.listen((event) {
+      if (!mounted) return;
+      event.whenOrNull<void>(
+        inputsUpdated: (inputs) {
+          _controller
+              .setTrack(
+                inputs.camera.isEnabled ? widget.client.participants.local.media?.camera.track : null,
+              )
+              .whenComplete(() => setState(() {}));
+        },
+        participantUpdated: (participant) {
+          if (participant.id != widget.client.participants.local.id) return;
```

Comment:
In an earlier event handler I think we checked if the participant was local by using `participant.info.isLocal`, but here we're accessing what looks to be a participant collection to compare IDs. Just curious about why we went with different ways to deduce whether this is a local participant?